car.js

```diff
@@ -135,10 +135,38 @@ async function writeStream (stream) {
   return new CarDatastore(reader, writer)
 }
 
+async function traverseBlock (block, get, car, concurrency = 1, seen = new Set()) {
+  const cid = await block.cid()
+  await car.put(cid, block.encodeUnsafe())
+  seen.add(cid.toString('base58btc'))
+  if (cid.codec === 'raw') return
+  const reader = block.reader()
+  const missing = link => !seen.has(link.toString('base58btc'))
+  const links = Array.from(reader.links()).filter(missing).map(([, link]) => link)
+  while (links.length) {
+    const chunk = links.splice(0, concurrency)
+    const blocks = chunk.map(get)
```

Comment:
ah, so this is the point at which `concurrency` comes in to play. So the docs are going to have to make it clear that `concurrency` only applies on a links-per-block basis and not across all blocks. It's only really helpful if you have blocks with a lot of links in them.

Reply:
ya, i tried to think of a variable name that was more descriptive here but couldn’t come up with much other than `readBatchSize` and that seemed pretty ugly.